tools/plugins/com.liferay.ide.upgrade.commands.ui/src/com/liferay/ide/upgrade/commands/ui/internal/code/ConfigureWorkspaceProductKeyCommand.java

```diff
@@ -118,6 +151,118 @@ public IStatus validate(Object[] selection) {
 
 	}
 
+	private void _downloadReleaseApiJar(String productKey) throws URISyntaxException {
+		if (productKey == null) {
+			return;
+		}
+
+		String targetplatformVersion = _getGradleWorkspaceTargetPlatform();
+
+		if (Objects.isNull(targetplatformVersion)) {
+			return;
+		}
+
+		String productType = productKey.substring(0, productKey.indexOf("-"));
+
+		Job job = new Job("download release api job") {
+
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				try {
+					_downloadReleaseCompileOnlyPom(targetplatformVersion, productType);
+				}
+				catch (Exception e) {
+				}
+
+				return Status.OK_STATUS;
+			}
+
+		};
+
+		job.addJobChangeListener(
+			new JobChangeAdapter() {
+
+				@Override
+				public void done(IJobChangeEvent event) {
+					try {
+						String releaseApiUrl = _getDownloadReleaseApiUrl(targetplatformVersion, productType);
+
+						if (Objects.isNull(releaseApiUrl)) {
+							LiferayCore.logError("Failed to get release api download url");
+
+							return;
+						}
+
+						File cacheDir = new File(
+							System.getProperty("user.home"), ".liferay-ide/release-api/" + productKey);
+
+						HttpUtil.download(new URI(releaseApiUrl), cacheDir.toPath(), false);
```

Comment:
hey @simonjhy  and @Seiphon   I don't want us to download the release-api jar dynamically at run time.  I think we should just process the release-api jars for each major version and store the information in txt files in our plugin.  One for 7.0.x 7.0.10.x, 7.1.x, 7.1.10.x, etc (one for CE latest GA and latest DXP FP/SP per version).  Because this feature should not care the differnces between 7.10.2.2 and 7.10.2.fp22. Those can both use the same static text file (7.10.2.x-versions.txt), or whatever.

Reply:
@simonjhy do we have other comands that are in a non-ui plugin?

Reply:
We should add the major portion of this "command" code to the non-ui plugin so it can be unit tested without having to run functional tests only.